hipparchus-optim/src/test/java/org/hipparchus/optim/nonlinear/vector/leastsquares/SequentialGaussNewtonOptimizerWithQRTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Hipparchus project under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The Hipparchus project licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hipparchus.optim.nonlinear.vector.leastsquares;
+
+import java.io.IOException;
+
+import org.hipparchus.exception.LocalizedCoreFormats;
+import org.hipparchus.exception.MathIllegalStateException;
+import org.hipparchus.linear.QRDecomposer;
+import org.hipparchus.optim.LocalizedOptimFormats;
+import org.hipparchus.optim.SimpleVectorValueChecker;
+import org.hipparchus.optim.nonlinear.vector.leastsquares.LeastSquaresProblem.Evaluation;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * <p>Some of the unit tests are re-implementations of the MINPACK <a
+ * href="http://www.netlib.org/minpack/ex/file17">file17</a> and <a
+ * href="http://www.netlib.org/minpack/ex/file22">file22</a> test files.
+ * The redistribution policy for MINPACK is available <a
+ * href="http://www.netlib.org/minpack/disclaimer">here</a>/
+ *
+ */
+public class SequentialGaussNewtonOptimizerWithQRTest
+    extends AbstractSequentialLeastSquaresOptimizerAbstractTest {
+
```

Comment:
I don't see the difference between this test and SequentialGaussNewtonOptimizerWithQRNormalTest,
apart from some methods moved around and the expected exception check differently.
Shouldn't SequentialGaussNewtonOptimizerWithQRNormalTest be removed?